cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/query/IndexCreator.java

```diff
@@ -117,12 +117,16 @@ private String ensureIndexed(Index index) {
         try {
             if(index.indexName == null){
                 // generate a name for the index.
-                String indexName = this.generateIndexName();
+                String indexName = this.generateIndexName(listIndexesInDatabaseQueue());
                 if(indexName == null){
+                    logger.warning("Failed to generate unique index name");
                     return null;
                 }
 
-                index = Index.getInstance(index.fieldNames,indexName,index.indexType,index.indexSettings);
+                index = Index.getInstance(index.fieldNames,
+                                          indexName,
+                                          index.indexType,
+                                          index.indexSettings);
```

Comment:
hmmm I don't get the warning in studio, are you using the latest release of IDEA?